src/compiler/builder.ts

```diff
@@ -1182,7 +1182,8 @@ function getBuildInfo(state: BuilderProgramState, bundle: BundleBuildInfo | unde
         const { optionsNameMap } = getOptionsNameMap();
         for (const name of getOwnKeys(options).sort(compareStringsCaseSensitive)) {
             const optionInfo = optionsNameMap.get(name.toLowerCase());
-            if (optionInfo?.affectsBuildInfo) {
+            // all semantic diagnostics affect build info
+            if (optionInfo && (optionInfo.affectsBuildInfo || optionInfo.affectsSemanticDiagnostics)) {

```

Comment:
Lets not change this. You can assert instead that if `option.affectsBuildInfo` is not `true` then `affectsSemanticDiagnostics` is not true either

Reply:
Reverted and moved this to an assertion in a unit test.